src/cluster_asm.c

```diff
@@ -1096,38 +1088,7 @@ void asmSyncWithSource(connection *conn) {
         err = asmSendSlotRangesSync(conn, task);
         if (err) goto write_error;
 
-        task->state = ASM_SYNCSLOTS_REPLY;
-        return;
-    }
-
-    if (task->state == ASM_SYNCSLOTS_REPLY) {
```

Comment:
it seems a reply is clear?

Reply:
make sense, thanks

Reply:
yes, good catch, we should open another channel after confirmation from the main channel. Looks like we'll need the reply. Then, if we are not gaining much, I wonder if we should leave as it is and associate with client_id not to diverge a lot from replicaof. So, the newcomers can see something familiar.